azure/application-gateway/variables.tf

```
@@ -0,0 +1,207 @@
+variable "sku" {
+  description = "Properties relating to the SKU of the Applicaton Gateway"
+  type = object({
+    name     = string
+    tier     = string
+    capacity = optional(string)
+  })
+  default = {
+    name = "Standard_v2"
+    tier = "Standard_v2"
+  }
+}
+
+variable "autoscale" {
+  description = "Properties relating to the Autoscalling of the Applicaton Gateway"
+  type = object({
+    min = string
+    max = string
+  })
+  default = {
+    max = "3"
+    min = "1"
+  }
+}
```

Put the defaults into the `object({})` instead, as the current configuration will force the user to specify both where they may just want to specify the max so it's for example, 5 instead of 3, but keep min to 1.

```suggestion
variable "autoscale" {
  description = "Properties relating to the Autoscalling of the Applicaton Gateway"
  type = object({
    min = optional(number, 1)
    max = optional(number, 3)
  })
  default = {}
}
```